_overviews/toolkit/munit-asynchronous-tests.md

```diff
@@ -9,9 +9,72 @@ next-page: munit-resources
 
 {% include markdown.html path="_markdown/install-munit.md" %}
 
-<!--
-Tutorial about testing methods that return futures
-Also about configuration of the ContextExectution
+In Scala, it is common to define *asynchronous* methods as returning a `Future` of their result.
+For example, consider an asynchronous variant of a `square` method:
 
-See https://scalameta.org/munit/docs/tests.html#declare-async-test
- -->
\ No newline at end of file
+```scala
+import scala.concurrent.{ExecutionContext, Future}
+
+object MathLib:
+  def square(x: Int)(using ExecutionContext): Future[Int] =
+    Future(x * x)
+end MathLib
+```
+
+You may be tempted to test it as follows:
+
+```scala
+// Import the global execution context, required to call async methods
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import MathLib.*
+
+class MathLibTests extends munit.FunSuite:
+  test("square") {
+    assertEquals(9, square(3))
+    assertEquals(16, square(-4))
+  }
+end MathLibTests
+```
+
+However, it refuses to compile with the following error message:
+
+```
+[error] -- Error: MathLibTests.scala:21:30
+[error] 21 |    assertEquals(9, square(3))
+[error]    |                              ^
+[error]    |      Cannot prove that concurrent.Future[Int] <:< Int.
+```
+
+To address this kind of situation, MUnit offers special support for `Future`s.
+A test can itself return a `Future[Unit]`.
+MUnit will wait behind the scenes for the resulting `Future` to complete, failing the test if it is has a failed result.
+
+You can therefore write the test as follows:
```

Comment:
It uses “You”. Is this fine?

Reply:
From https://github.com/scalacenter/docs.scala-lang/issues/8 :

> Use `You` instead of `We` to write the tutorials. Use the present tense.

Reply:
Ah, I missed that change…

![image](https://user-images.githubusercontent.com/332812/201882329-4682aff9-8546-4695-a549-a44f3dd51b39.png)

Reply:
Jamie raised the point during the Spree that `We` is maybe too formal, and we found a few tutorials online, for instance [Python Beginner's Guide](https://wiki.python.org/moin/BeginnersGuide), that use `You`. So we decided to use `You`.

It is not completely settled though. Some other tutorials, for instance [Mdn Web doc](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Implementing_a_promise-based_API) use `We` extensively. I'll study more the question and raise the point during the next Scala Toolkit meeting.

For now I am okay with both.